Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2024 Miguel Marco. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Miguel Marco
+-/
+import Mathlib.Data.Set.Function
+import Mathlib.Data.Set.Functor
+import Mathlib.Lean.Expr.ExtraRecognizers
+
+/-!
+# Sets in subtypes
+
+This file defines notation for sets in a type pulled down to sets in a subtype, and sets in
+a subtype, lifted up to sets in the ambient type.
+
+It also provides some related lemmas for convenience.
+
+## Notation
+
+Let `α` be a `Type`, `A B : Set α` two sets in `α`, and `C : Set ↑A` a set in the subtype `↑A`.
+
+- `A ↓∩ B` denotes `(Subtype.val ⁻¹' B : Set A)` (that is, `{x : ↑A | ↑x ∈ B}`).
+- `↑C` denotes `Subtype.val '' C` (that is, `{x : α | ∃ y ∈ C, ↑y = x}`).
```

Comment:
You don't need to mention this notation, since the arrow is coming from a CoeHead instance in another file. You could move this explanation to that file's module doc.

It would be good to mention over there in the module doc and the coercion that there are lemmas about it in `Mathlib.Data.Set.Subset`.